dependency.go

```diff
@@ -1,29 +1,28 @@
 package response
 
 import (
+	"context"
 	"net/http"
 )
 
-type (
-	Factory interface {
-		GetFormatWriter() FormatWriter
-		CreateResponse(data any, statusCode int) *DataResponse
-		CreateInternalServerErrorResponse(err error) *DataResponse
-	}
-	FactoryAPI interface {
-		Factory
+type FormatWriter interface {
+	Write(w http.ResponseWriter, statusCode int, data any) error
+}
 
-		CreateUnprocessableEntityResponse(attributesErrors map[string][]string, message error)
-		CreateNotFoundEntityResponse(message error) *DataResponse
-		CreateErrorResponse(message error, statusCode int) *DataResponse
-	}
-	Handler interface {
-		Handle(f Factory, r *http.Request) *DataResponse
-	}
-	HandlerAPI interface {
-		Handle(f FactoryAPI, r *http.Request) *DataResponse
-	}
-	FormatWriter interface {
-		Write(data any, w http.ResponseWriter) error
-	}
-)
+type Factory interface {
+	FactoryAPI
+
+	FormatWriter() FormatWriter
+}
+
+type FactoryAPI interface {
+	CreateResponse(ctx context.Context, statusCode int, data any) *DataResponse
```

Comment:
the function must have in Factory because Factory is using for html format

Reply:
`Factory` has all methods from `FactoryAPI`

Reply:
it is not right for html format. html must use template and only CreateResponse. example: function CreateUnprocessableEntityResponse don't need for rendering template

Reply:
Done